pypesto/profile/profile.py

```diff
@@ -76,10 +77,12 @@ def parameter_profile(
     result:
         The profile results are filled into `result.profile_result`.
     """
+    # Copy the problem to avoid side effects
+    problem_profiling = copy.deepcopy(problem)
```

Comment:
```suggestion
    problem = copy.deepcopy(problem)
```

Avoids the danger of still modifying the wrong `problem`?